packages/sanity/src/hydration-test/event-listener-test.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Registry, {getCompName} from '@ui-autotools/registry';
+import chai, {expect} from 'chai';
+import sinonChai from 'sinon-chai';
+import {hydrate} from 'react-dom';
+import {FloodEvents} from './flood-events';
+import {overrideEventListeners} from './override-event-listeners';
+
+chai.use(sinonChai);
+
```

Comment:
I'd like an overall comment here explaining the approach since this isn't trivial